numpy/core/arrayprint.py

```diff
@@ -781,6 +789,16 @@ def __call__(self, x):
             return self.format % x.astype('i8')
 
 
+class SubArrayFormat(object):
+    def __init__(self, format_function):
+        self.format_function = format_function
+
+    def __call__(self, arr):
+        if arr.ndim <= 1:
+            return "[" + ", ".join(self.format_function(a) for a in arr) + "]"
+        return "[" + ", ".join(self.__call__(a) for a in arr) + "]"
```

Comment:
Just just to note,  probably a more exhaustive solution would recursively call `array2string` somehow (but with very large `max_line_width`?). But this looks fine for this PR.